src/node/node_state.h

```diff
@@ -372,6 +372,8 @@ namespace ccf
           setup_consensus(ServiceStatus::OPENING, false, endorsed_node_cert);
           setup_progress_tracker();
           setup_history();
+          if (endorsed_node_cert.has_value())
```

Comment:
Can this be set in `setup_history()` instead?

Reply:
I don't think that works well for recovery, but I will double check, it's a good point.

Reply:
As discussed, this isn't possible, but I've cleaned up the flow so the history is always created first, and the conditions for setting the endorsed cert are clearer.